tools/config/__init__.py

```diff
@@ -980,10 +980,6 @@ def _generate_bootloader_build(self, rom_memories):
                 yield region._replace(filename=self.target.header_format)
 
         if self.target.restrict_size is not None:
-            new_size = int(self.target.restrict_size, 0)
-            new_size = Config._align_floor(
-                start + new_size, self.sectors
-            ) - start
```

Comment:
style nit pick: could you remove the extra line before the `if self.target.app_offset:` line?